docs/dev/dev_adr.md

```diff
@@ -0,0 +1,111 @@
+# Architecture Decision Records
+
+The intention is to document deviations from a standard Model View Controller (MVC) design.
+
+## Pivoting Compliance View
+
+The view that was preferred for compliance would be devices on y-axis but the features on the x-axis. However, the x-axis (features) cannot be known when building the Django model. The model ends up looking like:
+
+| Device   | feature | Compliance |
+| -------- | ------- | ---------- |
+| nyc-rt01 | aaa     | True       |
+| nyc-rt01 | ntp     | False      |
+| nyc-rt01 | dns     | False      |
+| nyc-rt02 | aaa     | True       |
+| nyc-rt02 | dns     | False      |
+
+The expected view required expected is something like:
+
+| Device   | aaa   | dns   | ntp   |
+| -------- | ----- | ----- | ----- |
+| nyc-rt01 | True  | False | False |
+| nyc-rt02 | True  |       | False |
+
+In order to accommodate this, `django-pivot` is used, which greatly simplifies building the query. However, `django-pivot` requires the ability to "count" vs a boolean. Because of that, there is a "shadow" field created that sets to 0 if False, and 1 if True. This is enforced on the `save` method of the `ConfigCompliance` model.
+
+## Compliance View
+
+Important to understand `Pivoting Compliance View` first. There is additional context for how to handle bulk deletes. The logic is to find all of the `ConfigCompliance` data, given a set of `Device` objects. 
+
+Additionally, this makes use of the `alter_queryset` method, as at start time of the application all features are not necessarily set and needs to be a runtime query that sets the x and y axis correctly.
+
+The absence of data, meaning, a device that is does not have a feature, is the equivalent of a None.
+
+## Dynamic Application Features
+
+There are features within the application that can be turned on/off for backup, compliance, and intended. When they are toggled, this will update what is shown in:
+
+* Navigation
+* Jobs
+* Data Sources
+* Tables
+* Template Contents
+
+This is generally handled with a pattern similar to:
+
+```python
+jobs = []
+if ENABLE_BACKUP:
+    jobs.append(BackupJob)
+if ENABLE_INTENDED:
+    jobs.append(IntendedJob)
+if ENABLE_COMPLIANCE:
+    jobs.append(ComplianceJob)
+jobs.extend([AllGoldenConfig, AllDevicesGoldenConfig])
+```
+
+## Home View
+
+The Home view is generally based on the model `GoldenConfig`, however, in reality the view that shows up is based on the core `Device` model. This is because, when a device is included in the Dynamic Group, this does not mean that there is an entry in `GoldenConfig` yet. So there is nothing to see yet, such as the ability to click and run job on a device. It was confusing to users as to what was shown in the view vs what is in scope currently.
```

Comment:
```suggestion
The Home view is generally based on the model `GoldenConfig`; however, in reality the view that shows up is based on the core `Device` model. This is because, when a device is included in the Dynamic Group, this does not mean that there is an entry in `GoldenConfig` yet. So there is nothing to see yet, such as the ability to click and run job on a device. It was confusing to users as to what was shown in the view vs what is in scope currently.
```